packages/flutter/lib/src/material/list_tile.dart

```diff
@@ -630,57 +632,69 @@ class ListTile extends StatelessWidget {
     return dense ?? tileTheme.dense ?? theme.listTileTheme.dense ?? false;
   }
 
-  TextStyle _titleTextStyle(ThemeData theme, ListTileThemeData tileTheme) {
-    final TextStyle textStyle;
-    switch(style ?? tileTheme.style ?? theme.listTileTheme.style ?? ListTileStyle.list) {
-      case ListTileStyle.drawer:
-        textStyle = theme.useMaterial3 ? theme.textTheme.bodyMedium! : theme.textTheme.bodyLarge!;
-        break;
-      case ListTileStyle.list:
-        textStyle = theme.useMaterial3 ? theme.textTheme.titleMedium! : theme.textTheme.titleMedium!;
-        break;
+  TextStyle _titleTextStyle(ThemeData theme, ListTileThemeData tileTheme, ListTileThemeData defaults) {
+    final TextStyle textStyle = titleTextStyle
+      ?? tileTheme.titleTextStyle
+      ?? defaults.titleTextStyle!;
+    if (theme.useMaterial3) {
```

Comment:
I don't think we should short-circuit here; still need to copyWith the textColor. And although the denseLayout special case seems a bit odd, leaving it as it is will preserve backwards compatibility. Material 3 apps shouldn't specify isDense:true.